core/src/test/java/org/elasticsearch/rest/CorsAnyOriginIT.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.rest;
+
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.node.Node;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.ESIntegTestCase.ClusterScope;
+import org.elasticsearch.test.rest.client.http.HttpResponse;
+
+import static org.elasticsearch.http.netty.NettyHttpServerTransport.SETTING_CORS_ALLOW_ORIGIN;
+import static org.elasticsearch.http.netty.NettyHttpServerTransport.SETTING_CORS_ENABLED;
+import static org.elasticsearch.http.netty.cors.CorsHandler.ANY_ORIGIN;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasKey;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+
+/**
+ * Test case for CORS where http.cors.enabled=true but no
+ * origin literals or regexp is set.  In this case, the
+ * server should allow any origin.
+ */
+@ClusterScope(scope = ESIntegTestCase.Scope.SUITE, numDataNodes = 1)
+public class CorsAnyOriginIT extends ESIntegTestCase {
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        return Settings.builder()
+                   .put(Node.HTTP_ENABLED, true)
+                   .put(SETTING_CORS_ENABLED, true)
+                   .put(SETTING_CORS_ALLOW_ORIGIN, ANY_ORIGIN)
+                   .put(super.nodeSettings(nodeOrdinal)).build();
+    }
+
+    public void testThatCorsEnabledReturnsAnyOrigin() throws Exception {
+        HttpResponse response = httpClient().method("GET")
```

Comment:
@spinscale I did it more to maintain consistency with the CorsRegexIT test that was already in there, as well as the CorsNotSetIT.  But I can just as easily test the functionality by moving it to the netty transport tests.  I will do that for this and CorsOriginIT.